Apps/frontend/src/views/Oscilloscope.svelte

```diff
@@ -1,45 +1,39 @@
 <script>
-    import {onMount} from "svelte";
-    import CoordinateSystem from "./CoordinateSystem.svelte";
-    import SineWave from "./SineWave.svelte";
+  import { onMount, onDestroy } from "svelte";
+  import CoordinateSystem from "./CoordinateSystem.svelte";
+  import SineWave from "./SineWave.svelte";
+  import { CANVAS_WIDTH } from "../const";
 
-    const canvasWidth = 1000;
-    const canvasHeight = 500;
-    const numIntervalsX = 20;
-    const numIntervalsY = 10;
+  let waveElement;
+  let scaleY = 1; // 1V per horizontal line
+  let socket;
 
-    let waveElement;
+  onMount(() => {
+    socket = new WebSocket("ws://localhost:9000");
```

Comment:
When using ws://backend:9000, it uses docker's internal network and you don't need to expose port 9000 on the backend.